honeybee_radiance_command/gensky.py

```diff
@@ -0,0 +1,221 @@
+
+
+class Gensky(Command):
+    """gensky command."""
+
+    __slots__ = ('_ang', '_month', '_day', '_time', '_time_zone', '_solar_time',
+                 '_input')
+
+    def __init__(self, ang=None, month=None, day=None, time=None, time_zone=None,
+                 solar_time=False, options=None, output=None):
+        """Command.
+
+        Args:
+            ang: A tuple of altitude and azimuth values.
+            month: An integer representing the number of the month. Count starts from 01.
+            day: An integer representing the number of the day in a month. Count starts
+                from 01.
+            time: A float representing hour and minute.
+            time_zone: A three letter text representing the time zone.
+                Following are acceptable time zones with their corresponding hour
+                differences from Greenwhich mean time;
+                YST +9, PST +8, MST +7, CST +6, EST +5, GMT 0, CET -1, EET -2, AST -3,
+                GST -4, IST -5.5, JST -9, NZST -12, YDT +8, PDT +7, MDT +6, CDT +5,
+                EDT +4, BST -1, CEST -2, EEST -3, ADT -4, GDT -5, IDT -6.5, JDT -10,
+                NZDT -13
+            solar_time: A boolean to use local solar time. If set to True then the time
+                is preceded by '+' sign and local solar time is used instead of local
+                standard time.
+            options: Command options. It will be set to Radiance default values if not
+                provided by user.
+            output: File path to the output file (Default: None).
+        """
+        Command.__init__(self, output=output)
+        self.options = options
+        if ang:
+            self.options = ang
+            self._ang = ang
+            self._month = None
+            self._day = None
+            self._time = None
+            self._time_zone = None
+            self._solar_time = None
+        else:
+            self._ang = None
+            self._month = month
+            self._day = day
+            self._time = time
+            self._time_zone = time_zone
+            self._solar_time = solar_time
+
+    @property
+    def options(self):
+        """Rpict options."""
+        return self._options
+
+    @options.setter
+    def options(self, value):
+        if not value:
+            value = GenskyOptions()
+
+        if not isinstance(value, GenskyOptions):
+            raise ValueError('Expected GenskyOptions not {}'.format(type(value)))
+
+        self._options = value
+
+    @property
+    def ang(self):
+        """Angles for altitude and azimuth."""
+        return self._ang
+
+    @ang.setter
+    def ang(self, value):
+        self._ang.value = value
+
+    @property
+    def month(self):
+        """Month."""
+        return self._month
+
+    @month.setter
+    def month(self, value):
+        if value and isinstance(value, int) and value in range(1, 13):
+            self._month = value
+        else:
```

Comment:
Let's use the `typing` module for all these checks. There is a function for int_in_range: https://github.com/ladybug-tools/honeybee-radiance-command/blob/master/honeybee_radiance_command/_typing.py

Reply:
This is a comment for all the setters that you have here. That way you don't have to repeat writing the same check with a similar message.